src/TraceEvent/Stacks/CallTree.cs

```diff
@@ -271,8 +272,17 @@ struct TreeCacheEntry
             public CallTreeNode Tree;
         }
 
-        private CallTreeNode FindTreeNode(StackSourceCallStackIndex stack)
+        private CallTreeNode FindTreeNode(StackSourceCallStackIndex stack, int depth = 0)
```

Comment:
@vancem I'm not particularly happy with the parameter name `depth`. Let me know if there's some other name you would prefer.

💭 For me the redeeming factor may be twofold:

1. The methods are private
2. The recursion guard immediately inside the method is super obvious

Reply:
I think 'callStackDepth' is better than 'depth'.